src/libraries/System.Private.Xml/src/System/Xml/Dom/XmlProcessingInstruction.cs

```diff
@@ -14,94 +14,93 @@ public class XmlProcessingInstruction : XmlLinkedNode
         private readonly string _target;
         private string _data;
 
-        protected internal XmlProcessingInstruction(string target, string data, XmlDocument doc) : base(doc)
+        protected internal XmlProcessingInstruction(string target, string? data, XmlDocument doc) : base(doc)
         {
-            _target = target;
-            _data = data;
-        }
+            ArgumentNullException.ThrowIfNull(target);
 
-        // Gets the name of the node.
-        public override string Name
-        {
-            get
+            if (target.Length == 0)
             {
-                if (_target != null)
-                    return _target;
-                return string.Empty;
+                throw new ArgumentException(SR.Xml_EmptyName);
             }
-        }
 
-        // Gets the name of the current node without the namespace prefix.
-        public override string LocalName
-        {
-            get { return Name; }
+            _target = target;
+            _data = data ?? string.Empty;
         }
 
-        // Gets or sets the value of the node.
+        /// <inheritdoc />
+        public override string Name => _target;
+
+        /// <inheritdoc />
+        public override string LocalName => Name;
+
+        /// <inheritdoc />
         [AllowNull]
         public override string Value
         {
-            get { return _data; }
-            set { Data = value!; } //use Data instead of data so that event will be fired
+            get => _data;
+            set { Data = value; } // use Data instead of data so that event will be fired and null will be normalized to empty string
```

Comment:
```suggestion
            set { Data = value; } // uses Data instead of data so that event will be fired and null will be normalized to empty string
```